cookbooks/aws-parallelcluster-install/resources/efa/partial/_setup.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+#
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+action :setup do
+  if efa_installed? && !::File.exist?(efa_tarball)
+    Chef::Log.warn("Existing EFA version differs from the one shipped with ParallelCluster. Skipping ParallelCluster EFA installation and configuration.")
+    return
+  end
+
+  # remove conflicting packages
+  # default openmpi installation conflicts with new install
+  # new one is installed in /opt/amazon/efa/bin/
+  package conflicting_packages do
+    action :remove
+    not_if efa_installed?
+  end
+
+  # update repos and install prerequisite packages
+  package_repos 'update package repos' do
+    action :update
+  end
+  package prereq_packages do
+    retries 3
+    retry_delay 5
+  end
+
+  action_download_and_install
+end
+
+
+action :download_and_install do
+  return if virtualized?
+
+  efa_tarball = "#{node['cluster']['sources_dir']}/aws-efa-installer.tar.gz"
+  efa_installer_url = "https://efa-installer.amazonaws.com/aws-efa-installer-#{node['cluster']['efa']['installer_version']}.tar.gz"
+
+  # Get EFA Installer
+  remote_file efa_tarball do
+    source efa_installer_url
+    mode '0644'
+    retries 3
+    retry_delay 5
+    not_if { ::File.exist?(efa_tarball) }
```

Comment:
Nitpick: you can use `action :create_if_missing` to this end.

Reply:
done, thanks!